### PR TITLE
a11y - add aria-label to external link instances

### PR DESCRIPTION
**Pull request by katiegoines**
#### Description of changes:
For any external links, added an aria-label to the link that includes the link text and (opens in new tab). This is so screen reader users know the link may not behave as normal.

Staging: 
To see on ExternalLink, see footer/GlobalNav/right nav: https://a11y-external-links-label.d1egzztxsxq9xz.amplifyapp.com/
To see on LinkCards: https://a11y-external-links-label.d1egzztxsxq9xz.amplifyapp.com/gen1/react/#:~:text=React%20Libraries%20on%20GitHub
To see on ExternalLinkButton: https://a11y-external-links-label.d1egzztxsxq9xz.amplifyapp.com/react/start/quickstart/
To see on contributor page: https://a11y-external-links-label.d1egzztxsxq9xz.amplifyapp.com/contribute/getting-started/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/components/ExternalLink/index.tsx [hunk 1]
```diff
@@ -15,10 +15,27 @@ const ExternalLink: React.FC<ExternalLinkProps> = ({
   href,
   className
 }) => {
+  const [label, setLabel] = React.useState('');
+
+  useEffect(() => {
+    const links = document.getElementsByTagName('a');
+    const externalButtons = Array.from(links).filter((link) => {
```
Comment (jacoblogan) [inline src/components/ExternalLink/index.tsx:22]: Nit: These are specifically NOT ExternalLinkButtons and are targetting just ExternalLinks?  If so can we change this variable name to `externalLinks` or something a bit more descriptive of what it is?

File: src/components/ExternalLink/index.tsx [hunk 2]
```diff
@@ -15,10 +15,27 @@ const ExternalLink: React.FC<ExternalLinkProps> = ({
   href,
   className
 }) => {
+  const [label, setLabel] = React.useState('');
+
+  useEffect(() => {
```
Comment (jacoblogan) [inline src/components/ExternalLink/index.tsx:20]: This pattern of iterating through all the links looking for the current one feels a little clunky.  Have you tried using a ref to accomplish this instead?  Something along the lines of 
```
const ExternalLink: React.FC<ExternalLinkProps> = ({
  children,
  href,
  className
}) => {
  const [label, setLabel] = React.useState('');
  const linkRef = useRef<HTMLAnchorElement>(null);

  useEffect(() => {
    if (linkRef.current) {
      setLabel(linkRef.current.innerText);
    }
  }, []);

  return (
    <a
      href={href}
      className={className}
      aria-label={label + ' (opens in new tab)'}
      rel="noopener noreferrer"
      target="_blank"
      onClick={() => {
        trackLink(href);
      }}
      ref={linkRef}
    >
      {children}
    </a>
  );
};
```

File: src/components/ExternalLink/index.tsx (hunk 1 above)
Comment (katiegoines) [reply src/components/ExternalLink/index.tsx]: Ah good catch. Thank yoU! That is absolutely not the right variable name.

File: src/components/ExternalLink/index.tsx (hunk 2 above)
Comment (katiegoines) [reply src/components/ExternalLink/index.tsx:21]: That's significantly less clunky, thank you.

**Review by jacoblogan — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Pull request merged**